### PR TITLE
regen: use route api if comma.internal is not accessible

### PR DESCRIPTION
**Pull request by fredyshox**

### CODE REVIEW

File: tools/lib/filereader.py [hunk 1]
```diff
@@ -1,13 +1,29 @@
...
+  hostname = urlparse(DATA_ENDPOINT).hostname
+  if not hostname:
+    return False
+
+  try:
+    _ = socket.gethostbyname(hostname)
+    return True
+  except OSError:
+    return False
+
```
Comment (adeebshihadeh) [inline tools/lib/filereader.py:19]: this can return None

Comment (fredyshox) [reply tools/lib/filereader.py:19]: good catch

Comment (fredyshox) [reply tools/lib/filereader.py:19]: hmm, if it throws anything other than OSError, this will be rethrown to the caller, which is expected

**Comment by adeebshihadeh** [comment]
I don't think this is really what we want. Now on my workstation, I go through a completely different code path than I would on my laptop at home. Even internally, we want to access files from the cloud too.

**Comment by fredyshox** [comment]
I could make it optional instead, by adding "--use-route-meta" flag to explicitly toggle that. 
The reason for doing that in the first place is that:
1. its useless for people outside
2. some files are available using CommaApi, while being unavailable at comma.internal

**Pull request closed without merging**